extensions/src/simple_example/index.ts

```diff
@@ -55,30 +55,30 @@ export default class SimpleTypescript extends extension(details, "ui", "customSa
     console.log(value);
   }
 
-  @reporter`Add ${{type: "string", defaultValue: "yee"}} to ${{type: "string", defaultValue: "haw"}}: strings simple`
-    simpleReporterString(x: string, y: string) {
-        return x + y;
-    }
+  @(scratch.reporter`Add ${{ type: "string", defaultValue: "yee" }} to ${{ type: "string", defaultValue: "haw" }}: strings simple`)
+  simpleReporterString(x: string, y: string) {
+    return x + y;
+  }
 
-    @reporter`Add ${{type: "number", defaultValue: 3}} to ${"number"}: number simple`
-    simpleReporterNumber(x: number, y: number) {
-        return x + y;
-    }
+  @(scratch.reporter`Add ${{ type: "number", defaultValue: 3 }} to ${"number"}: number simple`)
+  simpleReporterNumber(x: number, y: number) {
+    return x + y;
+  }
 
-    @reporter((instance, $) => $`Add ${{type: "string", defaultValue: "oo"}} to ${{type: "string", defaultValue: "wee"}}: strings complex`)
-    reporterWithCallbackString(x: string, y: string) {
-        return x + y;
-    }
+  @(scratch.reporter((instance, $) => $`Add ${{ type: "string", defaultValue: "oo" }} to ${{ type: "string", defaultValue: "wee" }}: strings complex`))
```

Comment:
TODO: Explore how the current implementation handles excess whitespace / new lines. E.g. what if this was formatted like this:

```
  @(scratch.reporter((instance, $) => $`Add 
    ${{ type: "string", defaultValue: "oo" }} to
    ${{ type: "string", defaultValue: "wee" }}: strings complex
  `))
```

Let's think about it, but imagine we should remove all new lines, and likely excess white space (so any consecutive spaces / tabs / new lines are reduced to a single space). This gives the developer freedom to format their templated string in a way that works for them without causing weird spacing in the look of their block.

Reply:
I believe this function effectively does this kind of thing (maybe not exactly the same, but fire it up in the typescript playground and see how it behaves): 

https://github.com/mitmedialab/prg-ai-storybook-web/blob/main/packages/authoring/dialogue.ts#L48